flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolder.java

```diff
@@ -164,6 +167,13 @@ public void lazyInitialize(
         context.lazyInitialize(
                 globalFailureHandler, mainThreadExecutor, operatorCoordinatorMetricGroup);
 
+        if (coordinator instanceof RecreateOnResetOperatorCoordinator) {
```

Comment:
Instead of passing checkpoint coordinator like this, would it be simpler to do the following:

- `OperatorCoordinatorHolder#lazyInitialize` can pass checkpoint coordinator to `context.lazyInitialize`.
- Add method `getCheckpointCoordinator` to `OperatorCoordinator#Context`, which can return the checkpoint coordinator obtained in the above step.
- `SourceCoordinatorContext#setIsProcessingBacklog` can use the checkpoint coordinator obtained from `operatorCoordinatorContext`.